online-shop/src/app/CartService.ts

```diff
@@ -0,0 +1,53 @@
+import { Observable, of } from "rxjs";
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from "@angular/core";
+import { IProduct } from "./IProduct";
+import { IProductWrapper } from "./IProductWrapper";
+
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class CartService {
+    cart : IProductWrapper[] = [];
+    readonly ROOT_URL = "http://localhost:3000/products";
```

Comment:
**suggestion:** you can keep the url either inside the `environment.ts` file or you can use a [proxy](https://angular.io/guide/build#proxying-to-a-backend-server)